packages/status-communities/src/chat.ts

```diff
@@ -1,13 +1,23 @@
 import { ChatMessage } from "./chat_message";
 import { chatIdToContentTopic } from "./contentTopic";
+import { createSymKeyFromPassword } from "./encryption";
 
+/**
+ * Represent a chat room. Only public chats are currently supported.
+ */
 export class Chat {
   private lastClockValue?: number;
   private lastMessage?: ChatMessage;
-  public id: string;
 
-  constructor(id: string) {
-    this.id = id;
+  private constructor(public id: string, public symKey: Uint8Array) {}
+
+  /**
+   * Create a public chat room.
+   */
+  public static async create(id: string) {
+    const symKey = await createSymKeyFromPassword(id);
```

Comment:
@cammellos does the generation of symmetric key look good? I'll be able to test properly once status desktop is released with the ability to create open communities.

Reply:
Hard to say without testing, 
at first glance, the kdf algorithm looks a bit different:
https://github.com/status-im/status-go/blob/26bfeddad483a649ef2f17e9729b9547f33979aa/waku/waku.go#L927 that's the paramters
https://cs.opensource.google/go/x/crypto/+/089bfa56:pbkdf2/pbkdf2.go;l=42 


This one is the one is being used in this code I believe:

https://github.com/cyrildever/ecies-geth/blob/eba67eba82bbbc54d54b13b815858be3a2ed34fe/lib/src/typescript/node.ts#L75

We can probably just run the two with the same input and check if they give different outputs

Reply:
ok no worries, I'll test it out and see :)